lib/rpc/handler/FallbackHandler.ts

```diff
@@ -88,4 +87,79 @@ export class FallbackHandler {
       providerId: getProviderId(chainId, providerName),
     }
   }
+
+  private async processAlarm(alarmEvent: AlarmEvent, metric: AWSMetricsLogger): Promise<HealthinessUpdate> {
+    let healthinessUpdate = new HealthinessUpdate(ProviderHealthiness.HEALTHY, ProviderHealthiness.HEALTHY)
+
+    if (alarmEvent.state === 'ALARM') {
+      healthinessUpdate = await this.updateDbItemWhenAlarm(alarmEvent)
+      if (healthinessUpdate.isChanged()) {
+        metric.putMetric(`RPC_GATEWAY_FALLBACK_${alarmEvent.providerId}_INTO_UNHEALTHY`, 1, MetricLoggerUnit.Count)
+        this.log.error(
+          `${alarmEvent.providerId} becomes UNHEALTHY due to ${alarmEvent.previousState}=>ALARM in ${alarmEvent.alarmName}`
+        )
+      }
+    } else if (alarmEvent.state === 'OK') {
+      healthinessUpdate = await this.updateDbItemWhenOk(alarmEvent)
+      if (healthinessUpdate.isChanged()) {
+        metric.putMetric(`RPC_GATEWAY_FALLBACK_${alarmEvent.providerId}_INTO_HEALTHY`, 1, MetricLoggerUnit.Count)
+        this.log.error(
+          `${alarmEvent.providerId} becomes HEALTHY due to ${alarmEvent.previousState}=>OK in ${alarmEvent.alarmName}`
+        )
+      }
+    }
+    return healthinessUpdate
+  }
+
+  private async updateDbItemWhenAlarm(alarmEvent: AlarmEvent): Promise<HealthinessUpdate> {
```

Comment:
it is strange for these two functions to be separate, instead of just using `alarmEvent.state` and doing the selection with the two functions

Reply:
Yes the logic you mentioned is actually in `processAlarm()`. The reason for such is to easy unit testing because the `AWSMetricsLogger` is somewhat not very easy to mock in UT if you are not using the global `metric` object, in my case. So I want to add another layer to not let it bother me in writing UT.

Reply:
Changing logic for testing is not optimal; You can just create a function thats called updateItemWhenEvent and pass that in right?